storm-core/src/jvm/org/apache/storm/metric/SystemBolt.java

```diff
@@ -156,7 +219,103 @@ private void registerMetrics(TopologyContext context, Map<String, String> metric
 
     @Override
     public void execute(Tuple input) {
-        throw new RuntimeException("Non-system tuples should never be sent to __system bolt.");
```

Comment:
can we check that the incoming stream is METRICS_STREAM and throw same exception if its not?